clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java

```diff
@@ -160,11 +185,9 @@ public FSDataOutputStream create(Path cPath, FsPermission permission, boolean ov
   @Deprecated
   public FSDataOutputStream createNonRecursive(Path cPath, FsPermission permission,
       boolean overwrite, int bufferSize, short replication, long blockSize, Progressable progress)
-      throws IOException {
-    TachyonURI path = new TachyonURI(Utils.getPathWithoutScheme(cPath.getParent()));
-    if (!mTFS.exist(path)) {
-      throw new FileNotFoundException("Parent directory does not exist!");
-    }
+          throws IOException {
+    TachyonURI parentPath = new TachyonURI(Utils.getPathWithoutScheme(cPath.getParent()));
+    tryOpen(parentPath);
```

Comment:
Is it safe to throw the more general `IOException` here (previous was `FileNotFound`)?